### DISCUSSION TITLE
Testing Mini Apps

### DISCUSSION THREAD

**Discussion by Kavinsky1337** [aiogram/aiogram#1464] [General]
So I decided to launch an http:// site while developing a mini app. To do this, I created a test account from the debug menu and in it I created a new bot. using the new token in the test environment I get the error raise **_TelegramUnauthorizedError(method=method, message=description)
_aiogram.exceptions.TelegramUnauthorizedError: Telegram server says - Unauthorized__**

How should I deal with this problem and is there a video that tells how to test bots in a test environment?

**Comment by LexGolubkin** [comment]
Same problem. I had a trouble with register test server, and now is that. May be telegram test server just does not registered API Keys in their Keys list or we should connect to another domain. But I can't find any information about another domain.

**Comment by nigma888** [comment]
Hi, I faced the same problem. The solution is quite simple. You need to create a session with api=TEST (an instance of the TelegramAPIServer class) and then pass the session into your bot.
example:
```
from aiogram import Bot
from aiogram.client.session.aiohttp import AiohttpSession
from aiogram.client.telegram import TEST

session = AiohttpSession(api=TEST)
bot = Bot(token=BOT_TOKEN, session=session)
```
